src/tools/peptide-search/config/PeptideSearchFormData.ts

```diff
@@ -0,0 +1,66 @@
+import { FormParameters } from '../types/peptideSearchFormParameters';
+
+export type SelectedTaxon = { label: string; id: string };
+
+export enum PeptideSearchFieldTypes {
+  textarea,
+  select,
+  autocomplete,
+}
+
+export type PeptideSearchFormValue = {
+  fieldName: string;
+  selected?: string | SelectedTaxon[] | boolean | number;
+  type?: PeptideSearchFieldTypes;
+  values?: Readonly<
+    Array<{ label?: string; value?: string | boolean | number }>
+  >;
+};
+
+export enum PeptideSearchFields {
+  peps = 'Peptide sequences',
+  taxIds = 'Taxons',
+  lEQi = 'Treat isoleucine and leucine as equivalent',
+  spOnly = 'Search UniProt Reviewed (Swiss-Prot) only',
+  name = 'Name',
+}
+
+export type PeptideSearchFormValues = Record<
+  PeptideSearchFields,
+  Readonly<PeptideSearchFormValue>
+>;
+
+const formData: Readonly<PeptideSearchFormValues> = Object.freeze({
```

Comment:
Can you remind me why the `Object.freeze`?

Reply:
To make sure to not mutate the object while processing it in the form, this is due to the complexity of the form description object and the difficulty to track unexpected side effects related to accidentally mutating it